pkg/mons/restore_quorum.go

```diff
@@ -304,6 +310,29 @@ func getMonDetails(goodMon string, monEndpoints []string) ([]string, string, str
 	return badMons, goodMonPublicIp, goodMonPort, nil
 }
 
+func fetchGoodMon(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, monID string) error {
+	args := []string{"daemon", fmt.Sprintf("mon.%s", monID), "mon_status"}
+	out, err := exec.RunCommandInLabeledPod(ctx, clientsets, fmt.Sprintf("mon=%s", monID), "mon", "ceph", args, clusterNamespace, true)
+	if err != nil {
+		return fmt.Errorf("failed to run command ceph daemon mon.%s mon_status %v", monID, err)
+	}
+
+	type monStatus struct {
+		State string `json:"state"`
+	}
+	var monStatusOut monStatus
+	err = json.Unmarshal([]byte(out), &monStatusOut)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal mon stat output")
+	}
+
+	if monStatusOut.State == "leader" || monStatusOut.State == "peon" {
+		return nil
+	}
+
+	return fmt.Errorf("mon %s state is not leader/peon", monStatusOut.State)
```

Comment:
Have you observed other states? Seems like if we get any state, at least the mon is responding, and perhaps we can still proceed.

Reply:
no I have seen/tried to other states also based on [this](https://docs.ceph.com/en/latest/rados/troubleshooting/troubleshooting-mon/#most-common-monitor-issues) ceph doc

 ```doc
If the Monitor is out of the quorum, then its state will be one of the following: probing, electing or synchronizing. If the state of the Monitor is leader or peon, then the Monitor believes itself to be in quorum but the rest of the cluster believes that it is not in quorum.
 ```
seems like checking this should be the right thing. Thoughts?

Reply:
Got it, based on that doc, checking just for leader/peon sounds good.

Reply:
add mon id to err as well?

Reply:
```suggestion
return fmt.Errorf("mon %s in %s state but not in leader/peon state", monID,monStatusOut.State)
```
Do you meant to log like this or similar?

Reply:
yeah, your suggestion is more descriptive.